SwiftCompilerSources/Sources/Optimizer/PassManager/Context.swift

```diff
@@ -429,6 +429,33 @@ extension Type {
 //                          Builder initialization
 //===----------------------------------------------------------------------===//
 
+extension Instruction {
+  /// Returns self, unless self is a meta instruction, in which case the next
+  /// non-meta instruction is returned. Returns nil if there are no non-meta
+  /// instructions in the basic block.
+  fileprivate var skippingMetaInstructions: Instruction? {
+    switch self {
+    case is DebugValueInst, is AllocStackInst:
+      return self.next?.skippingMetaInstructions
+    default:
+      return self
+    }
+  }
+
+  fileprivate var locationSkippingMetaInstructions: Location {
+    let location = self.location
+    // If one of the locations has inline information, always keep the current
```

Comment:
This comment is kind of redundant, because it just translates the code below into English.
Instead it would be better to add a comment _why_ we want this behavior, because this is not obvious at all.